### PR TITLE
[Sol->Yul] Implement .creationCode.

### PR DESCRIPTION
**Pull request by ekpyron**
Depends on https://github.com/ethereum/solidity/pull/8856

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -1212,7 +1212,20 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 			solAssert(false, "Blockhash has been removed.");
 		else if (member == "creationCode" || member == "runtimeCode")
 		{
-			solUnimplementedAssert(false, "");
+			solUnimplementedAssert(member != "runtimeCode", "");
```
Comment (axic) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:1307]: Why not also support the `runtimeCode`?

Does it require more than `runtimeObjectName(contract)` as long as we assert immutables are not used?

Comment (ekpyron) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:1307]: Yep - I tried that, but the runtime object is a nested yul object inside the creation yul object and apparently we currently have scoping rules for yul objects that prevent you from accessing the inner yul object from a different yul object, so for supporting runtimeCode (at least) that would need to change.

Comment (axic) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:1307]: You mean #6739?

Comment (ekpyron) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:1307]: I didn't know there was an issue for that :-). But yeah, that one exactly :-).

**Review by mijovic — APPROVED** [review]

**Pull request merged**